### ISSUE TITLE
unitTests.UnicodeTests fail with python2

### ISSUE THREAD

**Issue by mcepl** [pyparsing/pyparsing#63]
With tarball ``pyparsing-2.3.1~test5.tar.gz`` I get this one test failing with Python 2:
```
$ python2 unitTests.py  -v
Beginning test of pyparsing, version 2.3.1
Python version 2.7.15 (default, May 21 2018, 17:53:03) [GCC]
.........................................................................................
>>>> Starting test UnicodeTests
[u'\u039a\u03b1\u03bb\u03b7\u03bc\u03ad\u03c1\u03b1', ',', u'\u03ba\u03cc\u03c3\u03bc\u03b5', '!']
<<<< End of test UnicodeTests


E..........................................................................................
>>>> Starting test UnicodeTests
[u'\u039a\u03b1\u03bb\u03b7\u03bc\u03ad\u03c1\u03b1', ',', u'\u03ba\u03cc\u03c3\u03bc\u03b5', '!']
<<<< End of test UnicodeTests


E..
======================================================================
ERROR: UnicodeTests
----------------------------------------------------------------------
Traceback (most recent call last):
  File "unitTests.py", line 94, in _runTest
    self.runTest()
  File "unitTests.py", line 3666, in runTest
    result = pp.Dict(pp.OneOrMore(pp.Group(key_value))).parseString(sample)
  File "/home/matej/build/python/python-pyparsing/pyparsing-2.3.1/pyparsing.py", line 1729, in parseString
    raise exc
ParseException: Expected "=" (at char 13), (line:1, col:14)

======================================================================
ERROR: UnicodeTests
----------------------------------------------------------------------
Traceback (most recent call last):
  File "unitTests.py", line 94, in _runTest
    self.runTest()
  File "unitTests.py", line 3666, in runTest
    result = pp.Dict(pp.OneOrMore(pp.Group(key_value))).parseString(sample)
  File "/home/matej/build/python/python-pyparsing/pyparsing-2.3.1/pyparsing.py", line 1729, in parseString
    raise exc
ParseException: Expected "=" (at char 13), (line:1, col:14)

----------------------------------------------------------------------
Ran 183 tests in 2.346s

FAILED (errors=2)
```
[Complete build log with python2 run for unitTests.py omitted](https://github.com/pyparsing/pyparsing/files/2732928/build.log)

**Comment by mcepl** [comment]
Talking about tarballs. How in the world, do you generate them? The one at https://github.com/pyparsing/pyparsing/releases/download/pyparsing_2.3.0/pyparsing-2.3.0.tar.gz does include neither ``pyparsing-2.3.1/simple_unit_tests.py`` nor ``pyparsing-2.3.1/unitTests.py``, both of which are included when I run plain ``python setup.py sdist``.

**Comment by ptmcg** [comment]
You fixed the MANIFEST.IN file to add these files, but only after 2.3.0 was pushed out. They will be there in 2.3.1.

**Comment by mcepl** [comment]
Hmm, with the generated tarball from the top of the ``master`` (commit ), I get this:
```
[    4s] ======================================================================
[    4s] ERROR: runTest (simple_unit_tests.TestLiteral) (test_spec=PpTestSpec(desc='Simple fa
il - parse an empty string', expr="xyz", text='', parse_fn='parseString', expected_list=None,
 expected_dict=None, expected_fail_locn=0))
[    4s] ----------------------------------------------------------------------
[    4s] Traceback (most recent call last):
[    4s]   File "simple_unit_tests.py", line 78, in runTest
[    4s]     print(pp.ParseException.explain(exc))
[    4s]   File "pyparsing.py", line 347, in explain
[    4s]     callers = inspect.getinnerframes(exc.__traceback__, context=depth)
[    4s]   File "pyparsing.py", line 272, in __getattr__
[    4s]     raise AttributeError(aname)
[    4s] AttributeError: __traceback__
[    4s]
[    4s] ======================================================================
[    4s] ERROR: runTest (simple_unit_tests.TestLiteral) (test_spec=PpTestSpec(desc='Simple fa
il - parse a mismatching string', expr="xyz", text='xyu', parse_fn='parseString', expected_li
st=None, expected_dict=None, expected_fail_locn=0))
[    4s] ----------------------------------------------------------------------
[    4s] Traceback (most recent call last):
[    4s]   File "simple_unit_tests.py", line 78, in runTest
[    4s]     print(pp.ParseException.explain(exc))
[    4s]   File "pyparsing.py", line 347, in explain
[    4s]     callers = inspect.getinnerframes(exc.__traceback__, context=depth)
[    4s]   File "pyparsing.py", line 272, in __getattr__
[    4s]     raise AttributeError(aname)
[    4s] AttributeError: __traceback__
[    4s]
[    4s] ======================================================================
[    4s] ERROR: runTest (simple_unit_tests.TestLiteral) (test_spec=PpTestSpec(desc='Simple fa
il - parse a partially matching string', expr="xyz", text='xy', parse_fn='parseString', expec
ted_list=None, expected_dict=None, expected_fail_locn=0))
[    4s] ----------------------------------------------------------------------
[    4s] Traceback (most recent call last):
[    4s]   File "simple_unit_tests.py", line 78, in runTest
[    4s]     print(pp.ParseException.explain(exc))
[    4s]   File "pyparsing.py", line 347, in explain
[    4s]     callers = inspect.getinnerframes(exc.__traceback__, context=depth)
[    4s]   File "pyparsing.py", line 272, in __getattr__
[    4s]     raise AttributeError(aname)
[    4s] AttributeError: __traceback__
[    4s]
[    4s] ======================================================================
[    4s] ERROR: runTest (simple_unit_tests.TestLiteral) (test_spec=PpTestSpec(desc='Fail - pa
rse a partially matching string by matching individual letters', expr={"x" "y" "z"}, text='xy
', parse_fn='parseString', expected_list=None, expected_dict=None, expected_fail_locn=2))
[    4s] ----------------------------------------------------------------------
[    4s] Traceback (most recent call last):
[    4s]   File "simple_unit_tests.py", line 78, in runTest
[    4s]     print(pp.ParseException.explain(exc))
[    4s]   File "pyparsing.py", line 347, in explain
[    4s]     callers = inspect.getinnerframes(exc.__traceback__, context=depth)
[    4s]   File "pyparsing.py", line 272, in __getattr__
[    4s]     raise AttributeError(aname)
[    4s] AttributeError: __traceback__
[    4s]
[    4s] ======================================================================
[    4s] FAIL: testGrammar (examples.antlr_grammar_tests.Test)
[    4s] ----------------------------------------------------------------------
[    4s] Traceback (most recent call last):
[    4s]   File "examples/antlr_grammar_tests.py", line 80, in testGrammar
[    4s]     pyparsingRules = antlr_grammar.antlrConverter(antlrGrammarTree)
[    4s]   File "examples/antlr_grammar.py", line 170, in antlrConverter
[    4s]     pyparsingRule = __antlrRuleConverter(pyparsingRules, antlrRule)
[    4s]   File "examples/antlr_grammar.py", line 153, in __antlrRuleConverter
[    4s]     rule = __antlrAlternativesConverter(pyparsingRules, antlrRule)
[    4s]   File "examples/antlr_grammar.py", line 114, in __antlrAlternativesConverter
[    4s]     rule = __antlrAlternativeConverter(pyparsingRules, antlrBlock.a1)
[    4s]   File "examples/antlr_grammar.py", line 131, in __antlrAlternativeConverter
[    4s]     assert ruleRef in pyparsingRules
[    4s] AssertionError
[    4s]
[    4s] ----------------------------------------------------------------------
[    4s] Ran 29 tests in 0.092s
[    4s]
[    4s] FAILED (failures=1, errors=4)
[    4s] W:(y)}}...)
[    4s] ['xx', 'y', 'xx', 'yy', 'xx', 'y', 'x', 'y', 'xxx', 'y']
[    4s]
[    4s] Match several words, skipping whitespace - OneOrMore({{W:(x) | W:(y)}}...)
[    4s] ['x', 'x', 'y', 'xx', 'y', 'y', 'xx', 'y', 'x', 'y', 'x', 'xx', 'y']
[    4s]
[    4s] Match words and numbers - show use of results names to collect types of tokens - One
OrMore({{W:(ABCD...) | integer}}...)
[    4s] ['sdlfj', 23084, 'ksdfs', 8234, 'kjsdlfkjd', 934]
[    4s] - alpha: ['sdlfj', 'ksdfs', 'kjsdlfkjd']
[    4s] - int: [23084, 8234, 934]
[    4s]
[    4s] Using delimitedList (comma is the default delimiter) - And(W:(ABCD...) [, W:(ABCD...
)]...)
[    4s] ['xxyx', 'xy', 'y', 'xxyx', 'yxx', 'xy']
[    4s]
[    4s] Using delimitedList, with ':' delimiter - Combine(W:(0123...) [: W:(0123...)]...)
[    4s] ['0A:4B:73:21:FE:76']
[    4s]
[    4s] A parse action that adds new key-values - OneOrMore({integer}...)
[    4s] [27, 1, 14, 22, 89]
[    4s] - ave: 30.6
[    4s] - max: 89
[    4s] - min: 1
[    4s] - sum: 153
[    4s]
[    4s] Match with results name - Literal("xyz")
[    4s] ['xyz']
[    4s] - value: 'xyz'
[    4s]
[    4s] Match with results name - using naming short-cut - Literal("xyz")
[    4s] ['xyz']
[    4s] - value: 'xyz'
[    4s]
[    4s] Define multiple results names - And({W:(ABCD...,ABCD...) "=" integer})
[    4s] ['range', '=', 5280]
[    4s] - key: 'range'
[    4s] - value: 5280
[    4s]
[    4s] Use transformString to convert simple markup to HTML - And({* | / | _ "(" !W:()) ")"
})
[    4s] Show in <B>bold</B>, <U>underscore</U>, or <I>italic</I> type
[    4s]
[    4s] Simple Word match - Word(W:(xy))
[    4s] ['xxyxxyy']
[    4s]
[    4s] Simple Word match of two separate Words - And({W:(x) W:(y)})
[    4s] ['xxxxx', 'yy']
[    4s]
[    4s] Simple Word match of two separate Words - implicitly skips whitespace - And({W:(x) W
:(y)})
[    4s] ['xxxxx', 'yy']
[    4s] error: Bad exit status from /var/tmp/rpm-tmp.njjc5I (%check)
```

**Comment by ptmcg** [comment]
You're right, I missed this in the Travis CI test logs. These unit tests use the new ParseException.explain() method. Because of differences in accessing exception traceback contents between Py2 and Py3, I have disabled these tests under Py2, and added notes that explain() is only supported on Py3.

**Comment by ptmcg** [comment]
Released in 2.3.1
